structuralcodes/codes/ec2_2004/_section_7_3_crack_control.py

```diff
@@ -0,0 +1,933 @@
+            The tensile zone is that parg of the section which is calculated
+            to be in tension just before the formation of the first crack.
+        sigma_s (float): is the absolute value of the maximum stress in MPa
+            permitted in the reinforcement immediately after the formation
+            of the crack. This may be taken as theyield strength of the
+            reinforcement, fyk. A lower value may, however, be needed to
+            satisfy the crack width limits according to the maximum
+            bar size of spacing (see 7.3.3 (2)).
+        fct_eff (float): is the mean value of the tensile strength in MPa of
+            the concrete effective at the time when the cracks may first be
+            expected to occur: fct,eff=fct or lower (fct(t)), is cracking
+            is expected earlier than 28 days.
+        _k (float): is the coefficient which allow for the effect of
+            non-uniform self-equilibrating stresses, which lead to a
+            reduction of restraint forces. Use 'k_crack_min_steel_area'
+            to compute it
+            k=1 for webs w<=300mm or flanges widths less than 300mm
+            k=0.65 for webs w>=800mm or flanges with widths greater than 800mm
+            Intermediate values may be interpolated.
+        kc (float): is a coefficient which takes account of the stress
+            distribution within the section immediately prior to cracking and
+            the change of the lever arm.
+
+    Returns:
+        float: the minimm area of reinforcing steel within the tensile
+            zone in mm2.
+
+    Raises:
+        ValueError: if _k value is not between 0.65 and 1 or kc is not
+            larger than 0 and lower than 1.
+    """
+    fct_eff = abs(fct_eff)
+
+    if A_ct <= 0:
+        raise ValueError(f'A_ct={A_ct} must be larger than 0')
+    if sigma_s < 0:
+        raise ValueError(f'sigma_s={sigma_s} must be equal or larger than 0')
+    if _k < 0.65 or _k > 1.0:
+        raise ValueError(f'_k={_k} must be between 0.65 and 1')
+    if kc > 1 or kc < 0:
+        raise ValueError(f'kc={kc} must be lower than 1 and larger than 0')
+
+    return kc * _k * fct_eff * A_ct / sigma_s
+
+
+def k(h: float) -> float:
+    """Is the coefficient which allow for the effect of
+    non-uniform self-equilibrating stresses, which lead to a
+    reduction of restraint forces.
+    k=1 for webs w<=300mm or flanges widths less than 300mm
+    k=0.65 for webs w>=800mm or flanges with widths greater than 800mm
+
+    EUROCODE 2 1992-1-1:2004, Eq. (7.1)
+
+    Args:
+        h (float): flange length or flange width in mm
+
+    Returns:
+        float: k coefficient value
+
+    Raises:
+        ValueError: if h is less than 0
+    """
+    if h < 0:
+        raise ValueError(f'h={h} cannot be less than 0mm')
+    if h <= 300:
+        return 1
+    if h < 800:
+        interpol = scipy.interpolate.interp1d((300, 800), (1, 0.65))
+        return (float)(interpol(h))
```

Comment:
Consider removing the parentheses around `float`, and consider removing also `float` since `interpol` here most likely returns a `float`.